model/metrics/clr-metrics.yaml

```diff
@@ -0,0 +1,166 @@
+groups:
+  - id: metric.clr.gc.collections.count
+    type: metric
+    metric_name: clr.gc.collections.count
+    brief: "Number of garbage collections that have occurred since the process started."
+    instrument: counter
+    unit: "{collection}"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.objects.size
+    type: metric
+    metric_name: clr.gc.objects.size
+    brief: "The number of bytes currently allocated on the managed GC heap. Fragmentation and other GC committed memory pools are excluded."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.allocations.size
+    type: metric
+    metric_name: clr.gc.allocations.size
+    brief: >
+      The number of bytes allocated on the managed GC heap since the process
+      started. The returned value does not include any native allocations.
+    instrument: counter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.committed_memory.size
+    type: metric
+    metric_name: clr.gc.committed_memory.size
+    brief: >
+      The amount of committed virtual memory for the managed GC heap, as
+      observed during the latest garbage collection.
+    note: >
+      Committed virtual memory may be larger than the heap size because it
+      includes both memory for storing existing objects (the heap size) and some
+      extra memory that is ready to handle newly allocated objects in the
+      future.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.heap.size
+    type: metric
+    metric_name: clr.gc.heap.size
+    brief: >
+      The heap size (including fragmentation), as observed during the latest
+      garbage collection.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.heap.fragmentation.size
+    type: metric
+    metric_name: clr.gc.heap.fragmentation.size
+    brief: >
+      The heap fragmentation, as observed during the latest garbage collection.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.duration
+    type: metric
+    metric_name: clr.gc.duration
```

Comment:
I think this one is fine, just leaving this here for consideration:

There are some [recommendations](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#instrument-naming) on naming and it says: 
> instrument that measures passage of time should be called entity.time

So maybe `clr.gc.time` would be better? However Java calls it [`gc.duraration`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmgcduration) as well.

Reply:
I'll update to align with that recommendation.

Reply:
I've decided to make this `clr.gc.pause.time` more explicit since it might be useful for `clr.gc.time` to be used later if we do collect the actual timings for each GC collection.